### PR TITLE
Fix juggling equation, performance with null check and logical ops etc

### PR DESCRIPTION
**Pull request by arthurkushman**
- juggling equation 
- performance is_null -> null check
- logical operations precedence
- execute() method with lost 3 parameter in CommandsHandle::triggerCommand()

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 composer.phar
 composer.lock
 vendor/
-build/
\ No newline at end of file
+build/
+.idea/
```
Comment (irazasyed) [inline .gitignore]: Please remove `.idea` it should be on your global ignores list.

**Comment by irazasyed** [comment]
Thanks for opening this PR. Everything looks good. Just please fix the `.gitignore` file and I'll be glad to merge this 👍

**Comment by irazasyed** [comment]
Thanks 👍

**Comment by arthurkushman** [comment]
Thank u for merging :-)

**Comment by holtkamp** [comment]
This tool saved me a lot of time fixing such things: https://github.com/FriendsOfPHP/PHP-CS-Fixer

Can provide a pull request, would it be considered?

**Pull request merged**